### PR TITLE
Removing unused, legacy python compatibility

### PR DESCRIPTION
**Pull request by bsipocz**
Over time the code has been removed for this python <3.3 compatibility workaround, time to remove it from the namespace, too.

### CODE REVIEW

**Comment by astrofrog** [comment]
We should maybe have a changelog entry in case any affiliated packages use this?

**Comment by bsipocz** [comment]
(no one seems to use this, at least not in public github repos: https://github.com/search?q=astropy.utils.compat+invalidate_caches+-filename%3Amisc.py&type=Code)

File: CHANGES.rst [hunk 1]
```diff
@@ -368,7 +368,10 @@ astropy.utils
 - ``diff_values()``, ``report_diff_values()``, and ``where_not_allclose()``
   utility functions are moved from ``astropy.io.fits.diff``. [#7444]
 
-astropy.visualization
+- ``invalidate_caches()`` has been removed from the
+  ``astropy.utils.compat`` namespace, use it directly from ``importlib``. [#7872]
+
+  astropy.visualization
 ^^^^^^^^^^^^^^^^^^^^^
```
Comment (astrofrog) [inline CHANGES.rst]: Can you fix the syntax for the title here?

Comment (bsipocz) [reply CHANGES.rst]: oops

**Pull request merged**